### ISSUE TITLE
Two mutation lines in library (feature request)

### ISSUE THREAD

**Issue by TopChelf** [cadon/ARKStatsExtractor#697]
In the library on the line at the top, there is only one mutation column. The column is the sum of matrilineal and patrilineal mutations. As a feature request, I would love to see one column for each side. 
In some of our breed plans (past 20 on one side), I look for my tames that have an open side to breed back to one with Zero on both or a wild. I cannot do this looking at my files in the Smart Breeding application, shy of opening each dino one by one. Having two columns would allow me to do this effectively and at a glance. 
A step further would be adding the feature in the Breeding Plan tab. Currently, only the "up to mutations" column allows you to control the breeding plan by a sum of mutations and does not separate by line. The solution here would be adding a second box, one for "up to [X] matri mutations," and "up to [Y] patri mutations." Perhaps another breeding mode could be Open Mutations (has fewer than 20 mutations on at least one side).  

Thanks!

**Comment by VolatilePulse** [comment]
First off, I'm trying to figure out where this information came from. You're the second person in less than 24 hours that has made a similar request for the same reason. Is there a new breeding article or YouTube video out? This is all wrong information ad I'd like to speak with the person who is releasing it so we can correct it.

That being said, I'm not going to go into all of the details behind it, but the only thing that matters to ark is the creatures mutation (not it's parents' mutation counters) and that one of the creatures must be less than 20/20 (a creature with 500,000/20 + 500,000/20 can still be mutated). I will be posting images shortly to disprove this theory of a creature that needs to have on ancestor with 0/20 or less. If you have any further question, please ask or join our Discord.

**Comment by VolatilePulse** [comment]
![](https://i.imgur.com/PnPlQEK.jpg)![](https://i.imgur.com/29CsDTp.jpg)![](https://i.imgur.com/HiCIGfv.jpg)

**Comment by cadon** [comment]
Like VolatilePulse explained, the number that is shown in the library is the number that has to be smaller than 20 for a mutation to happen. A creature that has a counter larger or equal than 20 can still have mutated offspring if it mates with a creature that has less than 20. But the number shown is the number you're looking for. The specific mutation-counter of the parents doesn't matter, only their sum, which is shown.

**Comment by TopChelf** [comment]
Dino A is a male, and has say 500/20 patri, and say 1/20 matri. Dino B is a female 0/20 0/20. 

The offspring will have 501/20 patri and 0/20 matri... and a mutation point may be observed, if occurred, on the matri side if yielded greater than 0. 

I get that. It’s the same as your photos, no difference. However, having the two columns would let me see how many counters a baby may have or had in a library list rather scrolling the breeding plan or ancestory. I can see which Dino’s got muts etc, as is observed with the sum. 
Current in the breed plan if I say up to 20 muts and weigh my stats, it will rule out, as intended, all tames above 20 muts regardless of stats... There are limited automatic features that encourage breeding back for mutations from full counters. So in your example even, the breed plan would skip over the 500k/500k dino if I used the current “up to x mutation” throttle. The columns suggested give manual method to take my wilds and sub 20’s to another. That is to say that 500k/500k is my best traded stat male, I may throttle my matri mutations to say 10.   The throttle wouldn’t be looking for females sub 10 on the matri side... it would take females (female passes matri muts to baby) where the sum of its muts is equal to or less than 10 and offer that against the inputted throttle prioritizing by the weighted stats desired. Thus, offering a chance at mutation with the best pair that allows for counters to reset on one side. 

Having done this now, the bred offspring would have a large partri side, and remaining counters on the matri, IE 0-11 now. Yes, when bred it will pass all of them to the respective side by sex, but now I can see at a glance what my offspring from the OG parent is without checking ancestory also, etc. I would see this by finding my 1mil patri and say 11 matri dino, rather 1mil 11 sum.. In practice, my best Dino’s would continue to have one open side, as they’re always being bred back to something that has open counters on both sides, when breeding for muts over stats. 

Among a few other helpful reasons, my request again would be having a view toggle for separating the sum. If the request is ill recieved I’ll pull the post. 
Thanks.

**Comment by VolatilePulse** [comment]
I think it would be best to leave this post for future reference to be honest. There is a lot of information here that is readily available on the web, especially pertaining to ASB.

Im going to do my best to cover your points and explain why such a feature isn't necessary. In regards to ark, the only mutation counters it takes into affect are the current mutation counters of both parents during the breeding process. The only stipulation in Ark is that one parent has not met or exceeded the maximum counter, 20/20 (all mutation counters should be referred to as X/20 to alleviate confusion as that is how it is displayed in Ark).

Going with the images I posted above, the fresh offspring, 64/20, only has a chance to provide a mutated stat if it is mated with a creature with 19/20 or less as it's mutation counter (as can be seen from the father who has both parental mutation counters exceeding the max but still provided a mutation). The mutation counter in game is non-representative of the actual number of mutations due to it increasing additionally with every generation. This number should not be used for determining if you should mate with this creature other than the if the counter is greater than 20/20, it must mate with a creature with a mutation counter of 19/20 or less.

During the stat selection process for mutations, the stat is randomly selected, just like normal breeding (could be the better or worse stat, it's a 55% chance to get the better one though), the game looks at both parents to see if a mutation is possible (one parent must be 19/20 or less), and assigns a mutation randomly between all of the already selected stats.

I have more data and plan on doing a write up to cover all of this information, but I have a male that exceeded 20/20 and still had one of it's stats mutated by a 0/20 female.

While observing mutations can be tricky during mass breeding, mutations aren't stored in game anywhere. The counter increasing is the closest to the mutation being stored anywhere and it's not accurate unless you use specific breeding tactics. ASB takes this into consideration by making it very easy to observe the stats in the library view. If a mutation occurred but happened on a worse stat, it may not be easily observable, however, if it occurs on a better stat, ASB will easily display it as it would be 2 Lw higher for that stat. 

TL;DR
The mutation counter is purely a logic check in Ark and the only numbers that matter are the current creatures mutation counter (not displayed directly in game but can be obtained by adding both mutation counters together under ancestry).

All creatures are able to be mutated as long as it mates with a creature with 19/20 or less for their mutation counter.

Parental mutation counters are not used in the logic check and are only displayed in game as an easy way to identify where the mutation came from (only applies if both mutation counters are below 20/20) and to calculate the current creatures mutation counter.

When a mutation occurs, it is visible in several ways assuming you have access to enough data. Specifically for stats, one stat is always increased by 2 levels (pending multiple mutations in one creature). If the +2 Lw is not visible, it's very likely it occurred on speed, an unused stat like oxygen or damage for diplos, or it applied to the worse stat.

I'm sure I didn't answer all of your points, so maybe I need you to clarify how, assuming a dad with 40/20 (20/20 + 20/20) could mutate with a mom 20/20 (20/20 + 0/20) or a dad with 20/20 (10/20 + 10/20) could mutate with a mom 20/20 (20/20 + 0/20)? Maybe I'm missing the point altogether, but it's hard to be theoretical with some of these scenarios as I think there is some confusion as to what numbers mean what. If you'd prefer, we can replace "/20" with "/§" to further eliminate confusion.

**Comment by VolatilePulse** [comment]
> However, having the two columns would let me see how many counters a baby may have or had in a library list rather scrolling the breeding plan or ancestory. I can see which Dino’s got muts etc, as is observed with the sum.

I think this is the key to trying to understand what you're looking for. Instead of showing the creatures current mutation counter, you want to display both the patrilineal and matrilineal mutation counter so you can observe if a mutation has been applied, correct?

The issue with the above has two scenarios. If ASB was to display the mutation counters that you entered (or imported from the export file) when adding the creature to the library, this mutation has already been added a parents mutation counter. This would serve no purpose other than to force a user to manually add both counters to figure out what the creatures actual mutation counter is.

The other scenario is that ASB displays the parent creatures' mutation counters in the library view, assuming you'd also want the current creature's mutation counter displayed as well. So, for the images as an example, the library entry for the child would look something like "48 + 15 = 64". This would easily show you both parents' mutation counters prior to the birth of the offspring as well the offspring's counter. You can figure out that the values don't line up and that 1 mutation was ascertained. While this may help decipher the stigma behind the mutation counters, it doesn't serve a purpose. The only thing that matters for breeding is the creatures' Lw of stats and if the creature's mutation counter is less than 20/20.just because you received a new mutation does not mean the stats will necessary reflect it nor does it mean it was placed in a stat you desire.

If your scenario is for quick observations during mass breeding cycles, it is best to monitor the stats as a controlled parent (male or female with 19/20 or less mutation counter) can force a mutation on any creature's stat.

**Comment by cadon** [comment]
I see there's an issue with the breeding-plan if a max mutation limit is set. This should only filter out pairings where both parents exceed this limit and not pairings where one parent is below the limit. I'll look into that.

The other issue (as far as I understood) is that you want to use the splitted mutation-counters for an easier finding of the offspring in the library and as a marker that these creatures are the ones you currently use for breeding (because they have one mutation-counter which is less than 20), correct?

**Comment by DeerHunter666** [comment]
@cadon , from my pespective you are correct. This is the issue/perspective I tried to raise in Discord. Do appreaciate the recent updated tooltips over the mut counter which helps Users to manually work out a breeding plan
